Source/Core/Core/PowerPC/GDBThread.cpp

```diff
@@ -0,0 +1,1350 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// Originally written by Sven Peter <sven@fail0verflow.com> for anergistic.
+
+#include "Core/PowerPC/GDBThread.h"
+
+#include "Common/ChunkFile.h"
+#include "Common/CommonFuncs.h"
+#include "Common/CommonTypes.h"
+#include "Common/StringUtil.h"
+#include "Common/Thread.h"
+#include "Common/Logging/Log.h"
+#include "Core/BootManager.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "Core/HW/CPU.h"
+#include "Core/PowerPC/PowerPC.h"
+#include "Core/PowerPC/Interpreter/Interpreter_FPUtils.h"
+#include "Core/PowerPC/JitInterface.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <signal.h>
+//#include <unistd.h>
+#ifdef _WIN32
+#include <ws2tcpip.h>
+#include <iphlpapi.h>
+#include <iphlpapi.h>
+#else
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <netinet/in.h>
+#define SD_BOTH SHUT_RDWR
+#endif
+#include <stdarg.h>
+
+#define fail(msg)   \
+{                   \
+    DEBUG_LOG(GDB_THREAD, msg); \
+    gdb_deinit();   \
+    return;         \
+}
+
+#define failr(msg)  \
+{                   \
+    DEBUG_LOG(GDB_THREAD, msg); \
+    gdb_deinit();   \
+    return 0;       \
+}
+
+#define		GDB_MAX_BP	100
+
+#define		GDB_STUB_START	'$'
+#define		GDB_STUB_END	'#'
+#define		GDB_STUB_ACK	'+'
+#define		GDB_STUB_NAK	'-'
+
+#ifdef _WIN32
+#define SIGTRAP 5
+#define	SIGTERM 15
+#define SIGSTOP 23 // values listed for MIPS?
+#define SIGCONT 25 // values listed for MIPS?
+#ifndef MSG_WAITALL
+#define MSG_WAITALL  8
+#endif
+#endif
+
+#define REGISTER_ID(category, index) ((category << 8) | index)
+
+enum gdb_bp_type
+{
+    GDB_BP_TYPE_NONE = 0,
+    GDB_BP_TYPE_X,
+    GDB_BP_TYPE_R,
+    GDB_BP_TYPE_W,
+    GDB_BP_TYPE_A,
+};
+
+// --------------------------------------------------------------------------------------
+//  GDBThread Implementations
+// --------------------------------------------------------------------------------------
+GDBThread::GDBThread() :
+    previous_state(CPU::State::PowerDown),
+    sock(-1),
+    sig(0),
+    connected(false)
+{
+}
+
+GDBThread::~GDBThread()
+{
+    Terminate();
```

Comment:
We use 2 spaces for indentation.

Again, this can be fixed automatically by running clang-format, as described in the contributing guide (which I really suggest reading).